tests/elliptic_curves/util.py

```diff
@@ -3,29 +3,198 @@
+                positions["modulus"] - len(gradient.to_list()) - len(Q.to_list()),
+                positions["P"] - len(Q.to_list()),
+            ]
+        case (True, True, False):
+            expected = [[modulus], Q.to_list()]
+            positions_expected = [positions["modulus"] - len(gradient.to_list()) - len(P.to_list()), positions["Q"]]
+        case (True, True, True):
+            expected = [[modulus]]
+            positions_expected = (
+                [positions["modulus"] - len(gradient.to_list()) - len(P.to_list()) - len(Q.to_list())]
+                if P != Q
+                else [positions["modulus"] - len(gradient.to_list()) - len(P.to_list())]
+            )
+    expected = [*expected, (P_ + Q_).to_list()]
+    positions_expected = [*[el + len(P.to_list()) for el in positions_expected], len(P.to_list()) - 1]
+    expected = generate_extended_list(expected, positions_expected, 1)[1:]
+
+    return {
+        "unlocking_script": nums_to_script(unlocking_script),
+        "expected": generate_verify_from_list(expected),
+        "stack_elements": {
+            "lambda": StackNumber(
+                positions["lambda"], len(gradient.to_list()), False, roll if rolls["lambda"] else pick
+            ),
+            "P": StackEllipticCurvePoint(
+                StackNumber(positions["P"], len(P.x.to_list()), negations["P"], roll if rolls["P"] else pick),
+                StackNumber(
+                    positions["P"] - len(P.x.to_list()),
+                    len(P.x.to_list()),
+                    negations["P"],
+                    roll if rolls["P"] else pick,
+                ),
+            ),
+        }
+        if P == Q
+        else {
+            "lambda": StackNumber(
+                positions["lambda"], len(gradient.to_list()), False, roll if rolls["lambda"] else pick
+            ),
+            "P": StackEllipticCurvePoint(
+                StackNumber(positions["P"], len(P.x.to_list()), negations["P"], roll if rolls["P"] else pick),
+                StackNumber(
+                    positions["P"] - len(P.x.to_list()),
+                    len(P.x.to_list()),
+                    negations["P"],
+                    roll if rolls["P"] else pick,
+                ),
+            ),
+            "Q": StackEllipticCurvePoint(
+                StackNumber(positions["Q"], len(Q.x.to_list()), negations["Q"], roll if rolls["Q"] else pick),
+                StackNumber(
+                    positions["Q"] - len(Q.x.to_list()),
+                    len(Q.x.to_list()),
+                    negations["Q"],
+                    roll if rolls["Q"] else pick,
+                ),
+            ),
+        },
+    }
+
+
+def generate_tests(modulus, P, Q, positions: list[dict[str, int]]):  # noqa: N803
+    """Generate test cases starting from modulus, P, Q, and the positions in which the gradient, P and Q must be."""
+    test_cases = []
+
+    if P != Q:
+        for is_p_negated in [True, False]:
```

Comment:
what about for is_p_negated, ..., is_lambda_rolled in itertool.product([True, False], repeat = 5)?

Reply:
Done, thanks!